components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -3163,6 +3163,7 @@ public static AuthenticatedUser createAuthenticatedUser(String username, String
             }
         } else {
             authenticatedUser.setUserStoreDomain(userStoreDomain);
+            authenticatedUser.setFederatedIdPName(idpName);
```

Comment:
Since this should always be local idp we can set it as `FrameworkConstants.LOCAL_IDP_NAME`[1]

https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/02ddac20ac1dd59b064095935a97927891baa44c/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java#L3423